router/router.go

```diff
@@ -1117,9 +1117,19 @@ func (worker *workerT) postStatusOnResponseQ(respStatusCode int, respBody string
 
 		worker.rt.failedEventsChan <- *status
 
-		if respStatusCode >= 500 {
-			timeElapsed := time.Since(firstAttemptedAtTime)
-			if respStatusCode != types.RouterTimedOutStatusCode && respStatusCode != types.RouterUnMarshalErrorCode {
+		switch respStatusCode {
+		case transformer.DROP_STATUS_CODE:
+			status.JobState = jobsdb.Aborted.State
+		case types.RouterTimedOutStatusCode, types.RouterUnMarshalErrorCode:
+			addToFailedMap = false
+			addToAbortMap = false
+		case 429:
```

Comment:
[optional] just to make the code more readable.

```suggestion
		case http.StatusTooManyRequests:
```